### ISSUE TITLE
Crash when ending turn

### ISSUE THREAD

**Issue by nerochiaro** [wesnoth/wesnoth#3768]
I get random crashes when ending the current turn. If I reload the last autosave the crash does not usually happen again at the end of the turn.

Backtrace here: https://gist.github.com/nerochiaro/26822a0bbb8f513ffa1712b25ea4b9be

**Comment by jyrkive** [comment]
Your backtrace link gives a 404.

**Comment by nerochiaro** [comment]
> Your backtrace link gives a 404.

Fixed in the original report. Sorry.

**Comment by gfgtdf** [comment]
which version are you using?

**Comment by jyrkive** [comment]
The failing assert: https://github.com/wesnoth/wesnoth/blob/8160d9add4ae08d0dad1264285816c7fd7ca1a49/src/whiteboard/side_actions.cpp#L334

**Comment by gfgtdf** [comment]
github master ot 1.14 branch?


Also from the stacktrace it looks like it does _not_ happen on wne turn but rather when you execute a planned move (iirc that the 'h') hotkey, (and it is not the automatic execution of planned moves at turn end. (`allow_end_turn` in the code)). So ye what the the comment said, the code somehow thinks you are trying to exceut a move that was planned for a later turn. So yes maybe you remembre more? did you delete/reorder a lot of planned moves maybe?

**Comment by nerochiaro** [comment]
You are right, it was not the end of turn but when confirming a move (hotkey 'y'). I confused it with end of the turn because it was my last move. Changed the title of bug report accordingly.
The branch is the master one (i.e. the one you get by using the clone link on the main github page for the project.
The tutorial was the one with the quintains (1st one).
I do not recall creating any complicated sequence of moves. I am new to the game, still learning how to do that, in fact.

**Comment by jostephd** [comment]
@nerochiaro Could you try moving line 336 to above line 334 and running `./wesnoth --log-info=whiteboard`? It's a bit of a shot in the dark, but there's a chance that the output might have a clue for us.

You may know this, but just in case: you can disable planning mode by pressing `p`. This should avoid the bug until we fix it. And welcome to Wesnoth!

**Comment by jostephd** [comment]
Can we just change the assert @jyrkive points to a `return false;`, or would the fix be more involved than that?

**Comment by gfgtdf** [comment]
This was finally fixed
